vimscript.html.markdown

```diff
@@ -0,0 +1,572 @@
+---
+language: Vimscript
+filename: learnvimscript.md
+contributors:
+    - ["HiPhish", "http://hiphish.github.io/"]
+---
+
+## Introduction
+
+Here is a dirty secret: there is no such thing as a Vim scripting language.
+There are ex-commands, a subset of which implements features one would expect
+from a scripting language (variable bindings, functions, loops). This is what
+people usually mean when they speak of Vim script (also called VimL). Always
```

Comment:
My opinion, but in this sense vimL isn't less of a scripting language that bash is, and bash is considered a scripting language. Not to mention python and others have their interpreteres so how is it that they differ? The difference is just that the interpreter for vimL is vim itself, but it's still a scripting language. It even supports `self` for dictionary functions, lambdas etc making it much more complex than bash (a scripting language). Maybe you should say that vim Ex commands are integral part of the language, just as regular executable commands can be used in bash scripts, I think it would be more fitting.

Reply:
You are right, I think that was a bit too harsh.